### PR TITLE
feat: provide ui related custom configuration

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- Since we are able to configure the build to either display or not display some UI elements, we need to provide an object allowing us to consume it 

### Solutions

- Add a object providing UI feature flags 
- Either hide or show the UI elements allowing the user to create an account, based on the flag.

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6243179)(6243179)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6243179)(6243179)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6275446)(6275446)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6275446)(6275446)-staging-compat.apk)

**Review by vitorhugods — APPROVED** [review]
Maybe the kalium update is unnecessary?
In any case, LGTM

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/10/) **failed**.

File: buildSrc/src/main/kotlin/customization/FeatureConfigs.kt [hunk 1]
```diff
@@ -29,6 +29,10 @@ enum class FeatureConfigs(val value: String, val configType: ConfigType) {
     ALLOW_MARKETING_COMMUNICATION("allowMarketingCommunication", ConfigType.BOOLEAN),
     ALLOW_SSO("allowSSO", ConfigType.BOOLEAN),
     ALLOW_ACCOUNT_CREATION("allow_account_creation", ConfigType.BOOLEAN),
+    SUPPORT_URL("webSiteUrl", ConfigType.STRING),
+    ACCOUNTS_URL("accountsUrl", ConfigType.STRING),
+    BACKEND_URL("backendUrl", ConfigType.STRING),
+    BLACKLIST_HOST("blacklistHost", ConfigType.STRING),
```
Comment (vitorhugods) [inline buildSrc/src/main/kotlin/customization/FeatureConfigs.kt]: These do NOT exist anymore.

```suggestion
```

They are set in line `70`:

```kotlin
    DEFAULT_BACKEND_URL_ACCOUNTS("default_backend_url_accounts", ConfigType.STRING),
    DEFAULT_BACKEND_URL_BASE_API("default_backend_url_base_api", ConfigType.STRING),
    DEFAULT_BACKEND_URL_BASE_WEBSOCKET("default_backend_url_base_websocket", ConfigType.STRING),
    DEFAULT_BACKEND_URL_TEAM_MANAGEMENT("default_backend_url_teams", ConfigType.STRING),
    DEFAULT_BACKEND_URL_BLACKLIST("default_backend_url_blacklist", ConfigType.STRING),
    DEFAULT_BACKEND_URL_WEBSITE("default_backend_url_website", ConfigType.STRING),
    DEFAULT_BACKEND_TITLE("default_backend_title", ConfigType.STRING),
```

**Comment by vitorhugods** [comment]
⚠️ This also needs to be in `release/candidate`

File: buildSrc/src/main/kotlin/customization/FeatureConfigs.kt (hunk 1 above)
Comment (trOnk12) [reply buildSrc/src/main/kotlin/customization/FeatureConfigs.kt]: thanks

**Comment by vitorhugods** [comment]
> @trOnk12 looks like you are rolling back kalium to a previous commitish.
> 
> This means that the PR's target branch (develop) is using a newer version of Kalium, and the changes in this PR will rollback Kalium to an older version.
> develop 	This PR
> [6fb4f14c8eb75d92cdc6d9272a6abb25b973bce4](https://github.com/wireapp/kalium/tree/6fb4f14c8eb75d92cdc6d9272a6abb25b973bce4) 	[2329b49d3e8908c7a0fdfe5f6c952a230f9296cc](https://github.com/wireapp/kalium/tree/2329b49d3e8908c7a0fdfe5f6c952a230f9296cc)
> 
> Is this intentional?

It seems that this PR is switching kalium from `develop` to `release/candidate`.

If not needed, avoid updating the kalium reference to prevent this kind of thing.

**Comment by AndroidBob** [comment]
Build [12](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/12/) **failed**.

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-1664/13/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.2.0(6311742)(6311742)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.2.0(6311742)(6311742)-staging-compat.apk)

**Pull request merged**